pallets/xor-fee/src/lib.rs

```diff
@@ -840,15 +1191,75 @@ pub mod pallet {
             }
             Ok(().into())
         }
+
+        /// Allow use assets from white list to pay for fee
+        /// # Parameters:
+        /// - `origin`: caller
+        /// - `call`: call for which pay fee
+        /// - `asset_id`: asset in which pay fee, where None - XOR
+        #[allow(unused_variables)] // Used in extension
+        #[pallet::call_index(3)]
+        #[pallet::weight(<T as Config>::WeightInfo::xorless_call())]
```

Comment:
It is calculated inside extrinsic, and even if we can get weight of the provided `call` and use it in macro, this `call` may call another `call`, so it should be calculated like this

Reply:
It will be better to estimate the weight before the calling, it is a reason why we have these weight annotation before extrinsics.
If some call contains another calls - it is better to sum all of them. But I'm not sure how can we do that. 
If we can get at least the weight of provided call - we have to do it.